Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VTKExporter.h

```diff
@@ -32,24 +31,26 @@
 
 #include <sofa/core/objectmodel/lifecycle/RenamedData.h>
 
+#include <sofa/simulation/BaseSimulationExporter.h>
+
 namespace sofa::component::_vtkexporter_
 {
 
-class SOFA_COMPONENT_IO_MESH_API VTKExporter : public core::objectmodel::BaseObject
+class SOFA_COMPONENT_IO_MESH_API VTKExporter : public sofa::simulation::BaseSimulationExporter
 {
 public:
-    SOFA_CLASS(VTKExporter,core::objectmodel::BaseObject);
+    SOFA_CLASS(VTKExporter,sofa::simulation::BaseSimulationExporter);
 
 protected:
     sofa::core::topology::BaseMeshTopology* m_topology;
     sofa::core::behavior::BaseMechanicalState* m_mstate;
-    unsigned int m_stepCounter;
 
-    std::ofstream* outfile;
+    std::ofstream* m_outfile;
```

Comment:
```suggestion
    std::unique_ptr<std::ofstream> m_outfile;
```
I recommend a smart pointer